config/routes.rb

```diff
@@ -23,6 +23,10 @@
       resources :external_pledges,
                 only: [ :create, :update, :destroy ]
     end
+    get :weekly_report, :controller => :reports
```

Comment:
kinda hinted at this above, but I wonder if there's a way we can roll these up into a single endpoint, since they're all basically doing the same thing? Could take a date as an argument, like `time_report/:start_date` or something like that. Would have the benefit of making it more testable and maintainable going forward. What do you think?